crates/ide/src/syntax_highlighting/tests.rs

```diff
@@ -1230,6 +1232,94 @@ fn check_highlighting(
     expect.assert_eq(actual_html)
 }
 
+fn check_highlighting_with_config(
```

Comment:
Why did you embed `highlight_as_html()`? Compare to `check_highlighting()`. Make this normal and make `check_highlighting()` use this function.

Reply:
I'm not sure what you are suggesting. I think the problem is: how to make `highlight_as_html` use a custom highlighting configuration? It has the default configuration hardcoded into it?

Reply:
Then change it to take config. It's not complicated.